### PR TITLE
Detect BOM encoding in all file operations

### PR DESCRIPTION
**Pull request by sils**

### CODE REVIEW

**Comment by Mixih** [comment]
I think this has been moved to coala-utils....

**Comment by sils** [comment]
@Mixih look at the code!

**Comment by Mixih** [comment]
ugh, I feel stupid..

**Comment by sils** [comment]
who changed that regex? Since when are tags mandantory? That makes no sense!

**Comment by Mixih** [comment]
crap, the regex change was a bad merge that happened a few days ago. I'll look at reverting it

File: coalib/results/result_actions/OpenEditorAction.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 from coalib.results.Diff import Diff
 from coalib.results.Result import Result
 from coalib.results.result_actions.ResultAction import ResultAction
+from coalib.processes.Processing import detect_encoding
```
Comment (arush0311) [inline coalib/results/result_actions/OpenEditorAction.py]: did you mean coala_utils.FileUtils?

File: coalib/results/result_actions/ApplyPatchAction.py [hunk 2]
```diff
@@ -40,14 +42,18 @@ def apply(self,
...
             if not diff.delete:
                 new_filename = (diff.rename
                                 if diff.rename is not False
                                 else filename)
-                with open(new_filename, mode='w', encoding='utf-8') as file:
+                with open(new_filename, mode='w',
+                          encoding=detect_encoding(pre_patch_filename)) as file:
                     file.writelines(diff.modified)
 
+            if diff.delete or diff.rename:
```
Comment (adhikasp) [inline coalib/results/result_actions/ApplyPatchAction.py]: This doesn't seems relevant?

Comment (sils) [inline coalib/results/result_actions/ApplyPatchAction.py]: It is! If you delete the file before creating the renamed file you can't detect the encoding of the old file anymore.

**Comment by sils** [comment]
> This one is missed out
>
> https://github.com/coala/coala/blob/master/coalib/parsing/ConfParser.py#L51
>
> There also 2 'utf-8' in tests folder

Those are loading coafiles only, coafiles are our own file format and should be UTF 8.

**Comment by Asnelchristian** [comment]
ack https://github.com/coala/coala/pull/3387/commits/708bf6919126ebfe5fe6c860ddceac3f622447b0

**Comment by Asnelchristian** [comment]
@rultor merge

**Pull request merged**